tests/testserver/server.py

```diff
@@ -89,13 +89,14 @@ def _close_server_sock_ignore_errors(self):
 
     def _handle_requests(self):
         for _ in range(self.requests_to_handle):
-            sock = self._accept_connection()
-            if not sock:
-                break
+            with self._accept_connection() as sock:
+                sock = self._accept_connection()
+                if not sock:
+                    break
```

Comment:
delete line 93, it is redundant. I am not sure that `sock` can ever be None, so probably 94, 95 are not needed either.

Reply:
no, wrong, sorry.

Reply:
Can you try to run this with CPython? The code here is wrong: lines 92 and 93 clash. I think this should become
```
    def _handle_requests(self):
        for _ in range(self.requests_to_handle):
            try:
                ready, _, _ = select.select([self.server_sock], [], [], self.WAIT_EVENT_TIMEOUT)
                if not ready:
                    break
                sock = self.server_sock.accept()[0]
            except (select.error, socket.error):
                break
            try:
                handler_result = self.handler(sock)
                self.handler_results.append(handler_result)
            finally:
                sock.close()


```

Reply:
Done. Can't believe I missed that.